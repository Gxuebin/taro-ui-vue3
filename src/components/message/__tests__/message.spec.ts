import { mount } from '@vue/test-utils'
import AtMessage from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtMessage as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values },
  })
}

describe('AtMessage', () => {
  it('should render default AtMessage', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })
})
