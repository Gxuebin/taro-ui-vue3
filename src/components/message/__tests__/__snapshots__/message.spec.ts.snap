// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtMessage should render default AtMessage 1`] = `
<view
  class="at-message at-message--hidden at-message--info"
/>
`;

exports[`AtMessage should render prop -- class 1`] = `
<view
  class="test at-message at-message--hidden at-message--info test"
/>
`;

exports[`AtMessage should render prop -- style 1`] = `
<view
  class="at-message at-message--hidden at-message--info"
  style="color: red;"
/>
`;
